common/rdr/Exception.cxx

```diff
@@ -54,33 +54,15 @@ SystemException::SystemException(const char* s, int err_)
 {
   strncat(str_, ": ", len-1-strlen(str_));
 #ifdef _WIN32
-  // Windows error messages are crap, so use unix ones for common errors.
-  const char* msg = 0;
-  switch (err) {
-  case WSAECONNREFUSED: msg = "Connection refused";       break;
-  case WSAETIMEDOUT:    msg = "Connection timed out";     break;
-  case WSAECONNRESET:   msg = "Connection reset by peer"; break;
-  case WSAECONNABORTED: msg = "Connection aborted";       break;
-  }
-  if (msg) {
-    strncat(str_, msg, len-1-strlen(str_));
-  } else {
-#ifdef UNICODE
-    WCHAR* tmsg = new WCHAR[len-strlen(str_)];
-    FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
-                  0, err, 0, tmsg, len-1-strlen(str_), 0);
-    WideCharToMultiByte(CP_ACP, 0, tmsg, wcslen(tmsg)+1,
-		              str_+strlen(str_), len-strlen(str_), 0, 0);
-    delete [] tmsg;
-#else
-    char* currStr = str_+strlen(str_);
-    FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
-                  0, err, 0, currStr, len-1-strlen(str_), 0);
-#endif
-    int l = strlen(str_);
-    if ((l >= 2) && (str_[l-2] == '\r') && (str_[l-1] == '\n'))
-      str_[l-2] = 0;
-  }
+	wchar_t *currStr = new wchar_t[len-strlen(str_)];
+	FormatMessageW(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+                 0, err, 0, currStr, len-1-strlen(str_), 0);
+	WideCharToMultiByte(CP_UTF8, 0, currStr, wcslen(currStr), str_+strlen(str_),
+	                    len-1-strlen(str_), 0, 0);
+
+	int l = strlen(str_);
+	if ((l >= 2) && (str_[l-2] == '\r') && (str_[l-1] == '\n'))
+		str_[l-2] = 0;
```

Comment:
What does this part do? It wasn't in the existing code.

Reply:
It actually was there from the beginning. It removes trailing characters.

Reply:
I found something weird. Connection to a non existing display from the command line will give a error message but doing the same from the GUI will give an error message + trailing characters. I'll fix this tomorrow.